mx.py

```diff
@@ -7089,7 +7126,14 @@ def _returncode(status):
             return _returncode(status)
         remaining = end - time.time()
         if remaining <= 0:
-            abort('Process timed out after {0} seconds: {1}'.format(timeout, ' '.join(args)))
+            msg = 'Process timed out after {0} seconds: {1}'.format(timeout, ' '.join(args))
+            if nonZeroIsFatal:
+                abort(msg)
+            else:
+                log(msg)
+                for p, _ in _currentSubprocesses:
+                    _kill_process_group(p.pid, signal.SIGKILL)
+                return 1
```

Comment:
Maybe you can return something which is not a valid return code so it's possible to distinguish failed runs from timeouts? (e.g., `ERROR_TIMEOUT = 0x700000000 # not 32 bits`)